src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -463,4 +473,30 @@ public void setAllSubscribedPrefs(boolean b) {
 			e.printStackTrace();
 		}
 	}
+	
+	/**
+	 * Goes through all of the projects in session storage, determines if the user is currently
+	 * subscribed, and then writes that boolean to the "auto-subscribe" preferences.
+	 */
+	public void writeSubscribedProjects() {
+		System.out.println("Writing subscribed projects to auto-subscribe preferences...");
+		SessionStorage ss = PluginManager.getInstance().getDataManager().getSessionStorage();
+		Set<Long> subscribedIDs = ss.getSubscribedIds();
+		List<Project> projects = ss.getProjects();
+		Preferences pluginPrefs = InstanceScope.INSTANCE.getNode(Activator.PLUGIN_ID);
+		Preferences projectPrefs = pluginPrefs.node(PreferenceConstants.NODE_PROJECTS);
+		
+		for (Project p : projects) {
+			boolean subscribed = subscribedIDs.contains(p.getProjectID());
+			Preferences thisProjectPrefs = projectPrefs.node(p.getProjectID() + "");
+			thisProjectPrefs.putBoolean(PreferenceConstants.VAR_SUBSCRIBED, subscribed);
```

Comment:
This does make more sense. I don't think there's really any reason we need to be storing the false ones.